baseframe/forms/validators.py

```diff
@@ -161,6 +162,30 @@ def compare(self, value, other):
         return value != other
 
 
+class PublicEmailDomain(object):
+    def __init__(self, message=None):
+        self.message = message or _(u'The domain "{domain}" is not a public email domain.')
+
+    def __call__(self, form, field):
+        sniffedmx = mxsniff(field.data)
+        if len(sniffedmx['providers']) > 0:
+            return
```

Comment:
You need to check if any of the provider dictionaries has a key `'public'` with value `True`. The presence of a provider tells you nothing.